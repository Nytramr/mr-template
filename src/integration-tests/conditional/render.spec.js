import { Template } from '../../index';

describe('Template with conditionals', () => {
  test('simple conditional', async () => {
    const conditionalTpl =
      'Beginning of the file' +
      '\n' +
      '{{if .}} The value is truthy {{end}}' +
      '\n' +
      'End of the file' +
      '\n';

    const template = new Template('My template');
    expect(() => template.parse(conditionalTpl)).not.toThrow();

    expect(template.execute(true)).toMatchSnapshot();
    expect(template.execute(false)).toMatchSnapshot();
  });

  test('conditional with else', async () => {
    const conditionalTpl =
      'Beginning of the file' +
      '\n' +
      '{{if .}}' +
      '\n' +
      'The value is truthy' +
      '\n' +
      '{{else}}' +
      '\n' +
      'The value is falsy' +
      '\n' +
      '{{end}}' +
      '\n' +
      'End of the file' +
      '\n';

    const template = new Template('My template');
    expect(() => template.parse(conditionalTpl)).not.toThrow();

    expect(template.execute(true)).toMatchSnapshot();
    expect(template.execute(false)).toMatchSnapshot();
  });

  test('conditional with else if', async () => {
    const conditionalTpl =
      'Beginning of the file' +
      '\n' +
      '{{if eq . "a"}}' +
      '\n' +
      'The value is "a"' +
      '\n' +
      '{{else if eq . "b"}}' +
      '\n' +
      'The value is "b"' +
      '\n' +
      '{{else if eq . "c"}}' +
      '\n' +
      'The value is "c"' +
      '\n' +
      '{{else}}' +
      '\n' +
      'The value is other than "a", "b" or "c"' +
      '\n' +
      '{{end}}' +
      '\n' +
      'End of the file' +
      '\n';
    const template = new Template('My template');
    expect(() => template.parse(conditionalTpl)).not.toThrow();

    expect(template.execute('a')).toMatchSnapshot();
    expect(template.execute('b')).toMatchSnapshot();
    expect(template.execute('c')).toMatchSnapshot();
    expect(template.execute('z')).toMatchSnapshot();
  });
});
