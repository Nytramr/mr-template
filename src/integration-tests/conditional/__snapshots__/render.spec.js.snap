// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template with conditionals conditional with else 1`] = `
"Beginning of the file

The value is truthy

End of the file
"
`;

exports[`Template with conditionals conditional with else 2`] = `
"Beginning of the file

The value is falsy

End of the file
"
`;

exports[`Template with conditionals conditional with else if 1`] = `
"Beginning of the file

The value is "a"

End of the file
"
`;

exports[`Template with conditionals conditional with else if 2`] = `
"Beginning of the file

The value is "b"

End of the file
"
`;

exports[`Template with conditionals conditional with else if 3`] = `
"Beginning of the file

The value is "c"

End of the file
"
`;

exports[`Template with conditionals conditional with else if 4`] = `
"Beginning of the file

The value is other than "a", "b" or "c"

End of the file
"
`;

exports[`Template with conditionals simple conditional 1`] = `
"Beginning of the file
 The value is truthy 
End of the file
"
`;

exports[`Template with conditionals simple conditional 2`] = `
"Beginning of the file

End of the file
"
`;
